persistence/sql/persister_oauth2.go

```diff
@@ -161,7 +161,7 @@ func (r *OAuth2RequestSQL) toRequest(ctx context.Context, session fosite.Session
 
 // hashSignature prevents errors where the signature is longer than 128 characters (and thus doesn't fit into the pk).
 func (p *Persister) hashSignature(ctx context.Context, signature string, table tableName) string {
-	if table == sqlTableAccess && p.config.IsUsingJWTAsAccessTokens(ctx) {
+	if table == sqlTableAccess {
```

Comment:
Because that would break all existing tokens, and because that's not the reason why we hash:

>  // hashSignature prevents errors where the signature is longer than 128 characters (and thus doesn't fit into the pk).

This is only the case for JWT signatures which are typically > 128 chars.

I understand though why this is here, but we need to figure out either how to get the strategy in here, or how to solve this backwards compatible.

Reply:
fixed